src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -2,20 +2,26 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class MonikerProvider
     {
-        private readonly List<(Func<string, bool> glob, (string monikerRange, IEnumerable<Moniker> monikers))> _rules = new List<(Func<string, bool>, (string, IEnumerable<Moniker>))>();
+        private readonly List<(Func<string, bool> glob, (string monikerRange, IEnumerable<string> monikers))> _rules = new List<(Func<string, bool>, (string, IEnumerable<string>))>();
```

Comment:
why is this changed from `Moniker` to `string` again? Needed in #4621

Reply:
to move and hide moniker resolve logic inside `MonikerProvider`